### PR TITLE
Optimization: Introduced LED structs & Bug Fix

### PR DESCRIPTION
**Pull request by CedArctic**
**Changes:**
- Up to now, each key color was a single uint32_t variable of which the first 8 bits were unused. Also upon each columnCallBack() execution, the single uint32_t value had to be shifted and masked for each color to get the rgb variables. This commit creates an LED struct which contains 3 uint8_t variables to reduce memory and CPU usage.
- Bug fix: Some times when switching lighting profiles (especially animated ones) the animation timer would interfere with the lighting profile switch and lighting profiles would glitch. To counter this and make lighting profile switching snappier, interrupts while switching profiles have been disabled.

### CODE REVIEW

**Comment by Codetector1374** [comment]
Can you please remove the `LICENSE` file? Since that has already been merged as a separate PR.

File: source/light_utils.c [hunk 1]
```diff
@@ -26,15 +26,34 @@ static const uint8_t modKeyIDs[] = {0, 13, 14, 28, 40, 41, 42, 54, 55, 56, 57, 5
...
+        ledColors[modKeyIDs[i]].blue = blue;
     }
+}
+
+// Set specific key color
+void setKeyColor(led *key, uint32_t color){
+    key->red = (color >> 16) & 0xFF;
+    key->green = (color >> 8) & 0xFF;
+    key->blue = color & 0xFF;
 }
```
Comment (Codetector1374) [inline source/light_utils.c:59]: Please always add newline at the end of file

File: source/light_utils.h [hunk 2]
```diff
@@ -1,8 +1,17 @@
...
+
 /*
     Function Signatures
 */
-void setAllKeysColor(uint32_t* ledColors, uint32_t color);
-void setModKeysColor(uint32_t* ledColors, uint32_t color);
\ No newline at end of file
+void setAllKeysColor(led* ledColors, uint32_t color);
+void setModKeysColor(led* ledColors, uint32_t color);
+void setKeyColor(led *key, uint32_t color);
```
Comment (Codetector1374) [inline source/light_utils.h:17]: Same as above

File: source/light_utils.h [hunk 3]
```diff
@@ -1,8 +1,17 @@
 #include "board.h"
 #include "hal.h"
 
+/*
+    Structs
+*/
+// Struct defining an LED and its RGB color components
+typedef struct led{
+    uint8_t red, green, blue;
+}led;
```
Comment (Codetector1374) [inline source/light_utils.h:10]: Please name struct with _t. Also you don't have to use named struct if you are not referencing the struct within itself.

```C
typedef struct {
    uint8_t red;
    uint8_t green;
    uint8_t blue;
} led_t;
```

**Comment by CedArctic** [comment]
I've addressed all above requests. If there is anything else that needs to be done let me know.

**Pull request merged**